codespell_lib/_codespell.py

```diff
@@ -502,8 +527,20 @@ def extract_words(text, word_regex, ignore_word_regex):
     return word_regex.findall(text)
 
 
+def apply_uri_ignore_words(check_words, line, word_regex, ignore_word_regex,
+                           uri_regex, uri_ignore_words):
+    if not uri_ignore_words:
+        return
+    for uri in re.findall(uri_regex, line):
+        for uri_word in extract_words(uri, word_regex,
+                                      ignore_word_regex):
+            if "*" in uri_ignore_words or uri_word in uri_ignore_words:
+                check_words.remove(uri_word)
```

Comment:
I wonder if something even more like this might make more sense. Especially because your initial comment was people will just want to skip all typos in URIs wasn't it? Which means that 90% of the time we just want to hide the URI rather than process it twice.

```suggestion
    if "*" in uri_ignore_words:
      text = uri_regex.sub(' ', text)
    return word_regex.findall(text)


def apply_uri_ignore_words(check_words, line, word_regex, ignore_word_regex,
                           uri_regex, uri_ignore_words):
    for uri in re.findall(uri_regex, line):
        for uri_word in extract_words(uri, word_regex,
                                      ignore_word_regex):
            if uri_word in uri_ignore_words:
                check_words.remove(uri_word)
```

@larsoner do you have any strong feelings either way?